CONVENTIONS.md

```diff
@@ -129,6 +129,17 @@ it is used to manage all elements of the cluster.
   * Components that support workloads directly must not disrupt end-user workloads during upgrade or reconfiguration
     * E.g. the upgrade of a network plugin must serve pod traffic without disruption (although tiny increases in latency are allowed)
     * All components that currently disrupt end-user workloads must prioritize addressing those issues, and new components may not be introduced that add disruption
+* The platform should not disrupt workloads (reboot nodes) during normal operation
+  * If an admin requests a change to the system that has the clear expectation of disruption, the system may cause workload disruption (for example, an upgrade or machine configuration change should trigger a rolling reboot because that is expected)
+  * If an admin configures an optional mechanism like machine health checks or automatic upgrades, they are explicitly opting in to workload disruption and this constraint does not apply
```

Comment:
(I'd like to see us move towards encouraging automatic updates by default personally)